### PR TITLE
ORGS-205: Update GO SDK to support organization domains crud

### PR DESCRIPTION
**Pull request by NicolasLopes7**
This PR updates the go sdk to support the organization domains crud

### CODE REVIEW

File: organizationdomain/client.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+func NewClient(config *clerk.ClientConfig) *Client {
+	return &Client{
+		Backend: clerk.NewBackend(&config.BackendConfig),
+	}
+}
+
+type CreateParams struct {
+	clerk.APIParams
+	Name           string `json:"name" form:"name"`
```
Comment (gkats) [inline organizationdomain/client.go]: :wrench: What does the `form` tag do here? Same for all parameter fields.

File: organizationdomain/client.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+	OrganizationID string `json:"-" form:"-"`
```
Comment (gkats) [inline organizationdomain/client.go]: :wrench: This field should be removed. The organization ID is not part of the parameters to create an organization.

File: organizationdomain/client.go [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+	Backend clerk.Backend
+}
+
+func NewClient(config *clerk.ClientConfig) *Client {
+	return &Client{
+		Backend: clerk.NewBackend(&config.BackendConfig),
+	}
+}
+
+type CreateParams struct {
```
Comment (gkats) [inline organizationdomain/client.go:28]: :wrench: All fields should be pointers and omitted when empty. This is the convention used throughout the library. See the organization package for an example: https://github.com/clerk/clerk-sdk-go/blob/807c12461675e18bb88c5162b99cd3951e803902/organization/client.go#L32

File: organizationdomain/client.go [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+		return nil, err
+	}
+	req := clerk.NewAPIRequest(http.MethodPost, path)
+	req.SetParams(params)
+	domain := &clerk.OrganizationDomain{}
+	err = c.Backend.Call(ctx, req, domain)
+	return domain, err
+}
+
+type UpdateParams struct {
```
Comment (gkats) [inline organizationdomain/client.go:48]: :wrench: All fields should be pointers.

File: organizationdomain/client.go [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+	clerk.APIParams
+	OrganizationID string  `json:"-" form:"-"`
```
Comment (gkats) [inline organizationdomain/client.go]: :wrench: This field should be removed.

File: organizationdomain/client.go [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+	DomainID       string  `json:"-" form:"-"`
+	EnrollmentMode *string `json:"enrollment_mode" form:"enrollment_mode"`
+	Verified       *bool   `json:"verified" form:"verified"`
```
Comment (gkats) [inline organizationdomain/client.go]: :wrench: The `form` tag is meaningless here.

File: organizationdomain/client.go [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+}
+
+// Update updates an organization domain.
+func (c *Client) Update(ctx context.Context, params *UpdateParams) (*clerk.OrganizationDomain, error) {
```
Comment (gkats) [inline organizationdomain/client.go:57]: :wrench: The convention is to have Update methods accept the resource ID as a separate parameter. See here for an example: https://github.com/clerk/clerk-sdk-go/blob/807c12461675e18bb88c5162b99cd3951e803902/organization/client.go#L75

File: organizationdomain/client.go [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+	err = c.Backend.Call(ctx, req, res)
+	return res, err
+}
+
+type ListParams struct {
+	clerk.APIParams
+	clerk.ListParams
+	OrganizationID  string   `json:"-" form:"-"`
+	Verified        *bool    `json:"verified" form:"verified"`
+	EnrollmentModes []string `json:"enrollment_mode" form:"enrollment_mode"`
```
Comment (gkats) [inline organizationdomain/client.go]: :wrench: Same comment about using the `form` tag.

File: organization_domain.go [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+	ID                      string                                  `json:"id"`
+	OrganizationID          string                                  `json:"organization_id"`
+	Name                    string                                  `json:"name"`
+	EnrollmentMode          string                                  `json:"enrollment_mode"`
+	AffiliationEmailAddress *string                                 `json:"affiliation_email_address"`
+	Verification            *organizationDomainVerificationResponse `json:"verification"`
+	TotalPendingInvitations int                                     `json:"total_pending_invitations"`
+	TotalPendingSuggestions int                                     `json:"total_pending_suggestions"`
+	CreatedAt               int64                                   `json:"created_at"`
+	UpdatedAt               int64
```
Comment (gkats) [inline organization_domain.go]:
```suggestion
	UpdatedAt               int64 `json:"updated_at"`
```

File: organization_domain.go [hunk 10]
```diff
@@ -0,0 +1,29 @@
+package clerk
+
+type organizationDomainVerificationResponse struct {
```
Comment (gkats) [inline organization_domain.go]:
```suggestion
type OrganizationDomainVerification struct {
```

**Review by gkats — CHANGES_REQUESTED** [review]
Hey @NicolasLopes7 can you please take another look? We have to make sure that we follow the same conventions throughout the library. :smile:

**Comment by NicolasLopes7** [comment]
> Hey @NicolasLopes7 can you please take another look? We have to make sure that we follow the same conventions throughout the library. 😄

Ohh my bad! I thought I've left this one as a draft, was starting late tonight and would continue today's morning!

**Review by gkats — APPROVED** [review]

**Pull request merged**